corehq/apps/accounting/models.py

```diff
@@ -1304,6 +1304,17 @@ def change_plan(self, new_plan_version, date_end=None,
             funding_source=(funding_source or FundingSource.CLIENT),
             **kwargs
         )
+
+        # changing a plan overrides future subscriptions
+        future_subscriptions = Subscription.objects.filter(
+            subscriber=self.subscriber,
+            date_end=None,
```

Comment:
I think you can remove the ```date_end=None``` filter because we'd want to get rid of finite subscriptions as well.